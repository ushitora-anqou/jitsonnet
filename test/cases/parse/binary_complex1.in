1++1--1*1/~1%1<<1>>1<1|true^true==!true&true&&true||false
