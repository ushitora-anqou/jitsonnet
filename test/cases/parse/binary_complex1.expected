{ Syntax.v =
  (Syntax.Binary
     ({ Syntax.v =
        (Syntax.Binary
           ({ Syntax.v =
              (Syntax.Binary
                 ({ Syntax.v =
                    (Syntax.Binary
                       ({ Syntax.v =
                          (Syntax.Binary
                             ({ Syntax.v =
                                (Syntax.Binary
                                   ({ Syntax.v =
                                      (Syntax.Binary
                                         ({ Syntax.v =
                                            (Syntax.Binary
                                               ({ Syntax.v =
                                                  (Syntax.Number 1.);
                                                  loc =
                                                  { Syntax.fname =
                                                    "test/cases/parse/binary_complex1.in";
                                                    ran =
                                                    (Some { Syntax.startpos =
                                                            { Syntax.line = 1;
                                                              column = 0 };
                                                            endpos =
                                                            { Syntax.line = 1;
                                                              column = 1 }
                                                            })
                                                    }
                                                  },
                                                `Add,
                                                { Syntax.v =
                                                  (Syntax.Unary
                                                     (Syntax.Pos,
                                                      { Syntax.v =
                                                        (Syntax.Number 1.);
                                                        loc =
                                                        { Syntax.fname =
                                                          "test/cases/parse/binary_complex1.in";
                                                          ran =
                                                          (Some { Syntax.startpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    3 };
                                                                  endpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    4 }
                                                                  })
                                                          }
                                                        }));
                                                  loc =
                                                  { Syntax.fname =
                                                    "test/cases/parse/binary_complex1.in";
                                                    ran =
                                                    (Some { Syntax.startpos =
                                                            { Syntax.line = 1;
                                                              column = 2 };
                                                            endpos =
                                                            { Syntax.line = 1;
                                                              column = 4 }
                                                            })
                                                    }
                                                  }));
                                            loc =
                                            { Syntax.fname =
                                              "test/cases/parse/binary_complex1.in";
                                              ran =
                                              (Some { Syntax.startpos =
                                                      { Syntax.line = 1;
                                                        column = 0 };
                                                      endpos =
                                                      { Syntax.line = 1;
                                                        column = 4 }
                                                      })
                                              }
                                            },
                                          `Sub,
                                          { Syntax.v =
                                            (Syntax.Unary
                                               (Syntax.Neg,
                                                { Syntax.v =
                                                  (Syntax.Binary
                                                     ({ Syntax.v =
                                                        (Syntax.Binary
                                                           ({ Syntax.v =
                                                              (Syntax.Binary
                                                                 ({ Syntax.v =
                                                                    (
                                                                    Syntax.Number
                                                                    1.);
                                                                    loc =
                                                                    { Syntax.fname =
                                                                    "test/cases/parse/binary_complex1.in";
                                                                    ran =
                                                                    (Some 
                                                                    { Syntax.startpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    6 };
                                                                    endpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    7 } }) }
                                                                    },
                                                                  `Mult,
                                                                  { Syntax.v =
                                                                    (
                                                                    Syntax.Number
                                                                    1.);
                                                                    loc =
                                                                    { Syntax.fname =
                                                                    "test/cases/parse/binary_complex1.in";
                                                                    ran =
                                                                    (Some 
                                                                    { Syntax.startpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    8 };
                                                                    endpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    9 } }) }
                                                                    }));
                                                              loc =
                                                              { Syntax.fname =
                                                                "test/cases/parse/binary_complex1.in";
                                                                ran =
                                                                (Some 
                                                                { Syntax.startpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    6 };
                                                                  endpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    9 }
                                                                  })
                                                                }
                                                              },
                                                            `Div,
                                                            { Syntax.v =
                                                              (Syntax.Unary
                                                                 (Syntax.Lnot,
                                                                  { Syntax.v =
                                                                    (
                                                                    Syntax.Number
                                                                    1.);
                                                                    loc =
                                                                    { Syntax.fname =
                                                                    "test/cases/parse/binary_complex1.in";
                                                                    ran =
                                                                    (Some 
                                                                    { Syntax.startpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    11 };
                                                                    endpos =
                                                                    { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    12 } }) }
                                                                    }));
                                                              loc =
                                                              { Syntax.fname =
                                                                "test/cases/parse/binary_complex1.in";
                                                                ran =
                                                                (Some 
                                                                { Syntax.startpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    10 };
                                                                  endpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    12 }
                                                                  })
                                                                }
                                                              }));
                                                        loc =
                                                        { Syntax.fname =
                                                          "test/cases/parse/binary_complex1.in";
                                                          ran =
                                                          (Some { Syntax.startpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    6 };
                                                                  endpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    12 }
                                                                  })
                                                          }
                                                        },
                                                      `Mod,
                                                      { Syntax.v =
                                                        (Syntax.Number 1.);
                                                        loc =
                                                        { Syntax.fname =
                                                          "test/cases/parse/binary_complex1.in";
                                                          ran =
                                                          (Some { Syntax.startpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    13 };
                                                                  endpos =
                                                                  { Syntax.line =
                                                                    1;
                                                                    column =
                                                                    14 }
                                                                  })
                                                          }
                                                        }));
                                                  loc =
                                                  { Syntax.fname =
                                                    "test/cases/parse/binary_complex1.in";
                                                    ran =
                                                    (Some { Syntax.startpos =
                                                            { Syntax.line = 1;
                                                              column = 6 };
                                                            endpos =
                                                            { Syntax.line = 1;
                                                              column = 14 }
                                                            })
                                                    }
                                                  }));
                                            loc =
                                            { Syntax.fname =
                                              "test/cases/parse/binary_complex1.in";
                                              ran =
                                              (Some { Syntax.startpos =
                                                      { Syntax.line = 1;
                                                        column = 5 };
                                                      endpos =
                                                      { Syntax.line = 1;
                                                        column = 14 }
                                                      })
                                              }
                                            }));
                                      loc =
                                      { Syntax.fname =
                                        "test/cases/parse/binary_complex1.in";
                                        ran =
                                        (Some { Syntax.startpos =
                                                { Syntax.line = 1; column = 0
                                                  };
                                                endpos =
                                                { Syntax.line = 1;
                                                  column = 14 }
                                                })
                                        }
                                      },
                                    `Lsl,
                                    { Syntax.v = (Syntax.Number 1.);
                                      loc =
                                      { Syntax.fname =
                                        "test/cases/parse/binary_complex1.in";
                                        ran =
                                        (Some { Syntax.startpos =
                                                { Syntax.line = 1;
                                                  column = 16 };
                                                endpos =
                                                { Syntax.line = 1;
                                                  column = 17 }
                                                })
                                        }
                                      }));
                                loc =
                                { Syntax.fname =
                                  "test/cases/parse/binary_complex1.in";
                                  ran =
                                  (Some { Syntax.startpos =
                                          { Syntax.line = 1; column = 0 };
                                          endpos =
                                          { Syntax.line = 1; column = 17 } })
                                  }
                                },
                              `Lsr,
                              { Syntax.v = (Syntax.Number 1.);
                                loc =
                                { Syntax.fname =
                                  "test/cases/parse/binary_complex1.in";
                                  ran =
                                  (Some { Syntax.startpos =
                                          { Syntax.line = 1; column = 19 };
                                          endpos =
                                          { Syntax.line = 1; column = 20 } })
                                  }
                                }));
                          loc =
                          { Syntax.fname =
                            "test/cases/parse/binary_complex1.in";
                            ran =
                            (Some { Syntax.startpos =
                                    { Syntax.line = 1; column = 0 };
                                    endpos = { Syntax.line = 1; column = 20 }
                                    })
                            }
                          },
                        `Lt,
                        { Syntax.v = (Syntax.Number 1.);
                          loc =
                          { Syntax.fname =
                            "test/cases/parse/binary_complex1.in";
                            ran =
                            (Some { Syntax.startpos =
                                    { Syntax.line = 1; column = 21 };
                                    endpos = { Syntax.line = 1; column = 22 }
                                    })
                            }
                          }));
                    loc =
                    { Syntax.fname = "test/cases/parse/binary_complex1.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 1; column = 0 };
                              endpos = { Syntax.line = 1; column = 22 } })
                      }
                    },
                  `Lor,
                  { Syntax.v =
                    (Syntax.Binary
                       ({ Syntax.v = Syntax.True;
                          loc =
                          { Syntax.fname =
                            "test/cases/parse/binary_complex1.in";
                            ran =
                            (Some { Syntax.startpos =
                                    { Syntax.line = 1; column = 23 };
                                    endpos = { Syntax.line = 1; column = 27 }
                                    })
                            }
                          },
                        `Xor,
                        { Syntax.v =
                          (Syntax.Binary
                             ({ Syntax.v =
                                (Syntax.Binary
                                   ({ Syntax.v = Syntax.True;
                                      loc =
                                      { Syntax.fname =
                                        "test/cases/parse/binary_complex1.in";
                                        ran =
                                        (Some { Syntax.startpos =
                                                { Syntax.line = 1;
                                                  column = 28 };
                                                endpos =
                                                { Syntax.line = 1;
                                                  column = 32 }
                                                })
                                        }
                                      },
                                    `Equal,
                                    { Syntax.v =
                                      (Syntax.Unary
                                         (Syntax.Not,
                                          { Syntax.v = Syntax.True;
                                            loc =
                                            { Syntax.fname =
                                              "test/cases/parse/binary_complex1.in";
                                              ran =
                                              (Some { Syntax.startpos =
                                                      { Syntax.line = 1;
                                                        column = 35 };
                                                      endpos =
                                                      { Syntax.line = 1;
                                                        column = 39 }
                                                      })
                                              }
                                            }));
                                      loc =
                                      { Syntax.fname =
                                        "test/cases/parse/binary_complex1.in";
                                        ran =
                                        (Some { Syntax.startpos =
                                                { Syntax.line = 1;
                                                  column = 34 };
                                                endpos =
                                                { Syntax.line = 1;
                                                  column = 39 }
                                                })
                                        }
                                      }));
                                loc =
                                { Syntax.fname =
                                  "test/cases/parse/binary_complex1.in";
                                  ran =
                                  (Some { Syntax.startpos =
                                          { Syntax.line = 1; column = 28 };
                                          endpos =
                                          { Syntax.line = 1; column = 39 } })
                                  }
                                },
                              `Land,
                              { Syntax.v = Syntax.True;
                                loc =
                                { Syntax.fname =
                                  "test/cases/parse/binary_complex1.in";
                                  ran =
                                  (Some { Syntax.startpos =
                                          { Syntax.line = 1; column = 40 };
                                          endpos =
                                          { Syntax.line = 1; column = 44 } })
                                  }
                                }));
                          loc =
                          { Syntax.fname =
                            "test/cases/parse/binary_complex1.in";
                            ran =
                            (Some { Syntax.startpos =
                                    { Syntax.line = 1; column = 28 };
                                    endpos = { Syntax.line = 1; column = 44 }
                                    })
                            }
                          }));
                    loc =
                    { Syntax.fname = "test/cases/parse/binary_complex1.in";
                      ran =
                      (Some { Syntax.startpos =
                              { Syntax.line = 1; column = 23 };
                              endpos = { Syntax.line = 1; column = 44 } })
                      }
                    }));
              loc =
              { Syntax.fname = "test/cases/parse/binary_complex1.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 1; column = 0 };
                        endpos = { Syntax.line = 1; column = 44 } })
                }
              },
            `And,
            { Syntax.v = Syntax.True;
              loc =
              { Syntax.fname = "test/cases/parse/binary_complex1.in";
                ran =
                (Some { Syntax.startpos = { Syntax.line = 1; column = 46 };
                        endpos = { Syntax.line = 1; column = 50 } })
                }
              }));
        loc =
        { Syntax.fname = "test/cases/parse/binary_complex1.in";
          ran =
          (Some { Syntax.startpos = { Syntax.line = 1; column = 0 };
                  endpos = { Syntax.line = 1; column = 50 } })
          }
        },
      `Or,
      { Syntax.v = Syntax.False;
        loc =
        { Syntax.fname = "test/cases/parse/binary_complex1.in";
          ran =
          (Some { Syntax.startpos = { Syntax.line = 1; column = 52 };
                  endpos = { Syntax.line = 1; column = 57 } })
          }
        }));
  loc =
  { Syntax.fname = "test/cases/parse/binary_complex1.in";
    ran =
    (Some { Syntax.startpos = { Syntax.line = 1; column = 0 };
            endpos = { Syntax.line = 1; column = 57 } })
    }
  }